virtinst/devices/video.py

```diff
@@ -40,6 +40,8 @@ def default_model(guest):
         if guest.os.is_hvm():
             if guest.conn.is_qemu():
                 return "qxl"
+            if guest.supports_bochs_display():
+                return "bochs"
```

Comment:
In master, I moved the os.is_hvm() check to the top of the function, so this will collide a bit. But only because I think this should be tweaked a bit anyways

* Like you say, this won't ever be called. Your check should be moved before the is_qemu check. That qxl piece is meant to be a catchall for when the more specific cases above don't hit, but then not use qxl on xen.
* Rename supports_bochs_display to supports_video_bochs which will fit nicer if we add more of these helpers for video models in the future.
* The is_windows check should be hidden in a function like osinfo.supports_video_bochs().
* But I think guest.supports_bochs_display() should be dropped and instead opencode the logic here, so it's easier to follow the logic in a central place. The set_capabilities_defaults bit can be removed, it's not required. So here we would have `guest.is_uefi() and guest.osinfo.supports_video_bochs() and guest.lookup_domcaps().supports_bochs_display()`

Reply:
> In master, I moved the os.is_hvm() check to the top of the function, so this will collide a bit. But only because I think this should be tweaked a bit anyways
> 
> * Like you say, this won't ever be called. Your check should be moved before the is_qemu check. That qxl piece is meant to be a catchall for when the more specific cases above don't hit, but then not use qxl on xen.

Right.

> * Rename supports_bochs_display to supports_video_bochs which will fit nicer if we add more of these helpers for video models in the future.

Right.

> * The is_windows check should be hidden in a function like osinfo.supports_video_bochs().

I don't agree, Cole.
The **only** osinfo specific part is whether the guest is Windows or not. So,a `osinfo.supports_video_bochs()` would basically be `return not self.is_windows()` as the whole rest of the logic is part of the guest.

> * But I think guest.supports_bochs_display() should be dropped and instead opencode the logic here, so it's easier to follow the logic in a central place. The set_capabilities_defaults bit can be removed, it's not required. So here we would have `guest.is_uefi() and guest.osinfo.supports_video_bochs() and guest.lookup_domcaps().supports_bochs_display()`

Right!

Reply:
Re: the osinfo bit, the point is to just hide the osinfo specific logic inside a helper function there. Maybe it will grow in the future or need to be excluded for some certain distro some day, who knows. It's a minor distinction though so I'll leave it to your discretion

Though reading gerd's blog post I didn't find any mention of bochs + !windows, where does that bit of data come from?

Reply:
> Re: the osinfo bit, the point is to just hide the osinfo specific logic inside a helper function there. Maybe it will grow in the future or need to be excluded for some certain distro some day, who knows. It's a minor distinction though so I'll leave it to your discretion

I've adapted the patches according to your comments in the morning.
Yes, your point about expanding it in the future makes a lot of sense.

> 
> Though reading gerd's blog post I didn't find any mention of bochs + !windows, where does that bit of data come from?

So, Gerd never said explicitly it does not work on Windows. I've assumed that when I didn't see a Windows driver listed (as it's for QXL). But I also don't see one for VGA. :-/

I'll update the patches dropping that part.